### PR TITLE
docker.proxy.only_auth_for_admin - fix

### PR DESCRIPTION
**Pull request by kinjelom**

### CODE REVIEW

**Comment by jhunt** [comment]
Looks good to me.  Perhaps @kitsirota wants to take a look as well?

**Comment by kitsirota** [comment]
@wildloop thanks for this PR!  I'll review it today and post back with any notes/findings.

**Comment by kitsirota** [comment]
@wildloop I'll try to take a look today.  

The only other thing I could see that would break backwards compatibility is full access without auth on 443 as default (however adding only_auth_for_admin=false should be easy enough).  This is pretty common for preprod and testing environments and would be a bit of a pain to track down and fix on the CI side.

**Comment by wildloop** [comment]
default is (fix 2): `only_auth_for_admin: false` , so full access will require authentication, there is no case when full access is without authentication.

**Comment by kitsirota** [comment]
We need to support having TLS enabled and full access without auth, even if its a flag.  Otherwise this would break environments that are currently pushing containers without auth (eg CI testing)

**Comment by wildloop** [comment]
Sorry, I didn't know - fixed.

**Comment by wildloop** [comment]
I think that JS app will be available (future PR) on both ports on `/` location with auth same as `/v2/`

**Comment by kitsirota** [comment]
Awesome!  Thanks so much for the latest update.  I'll deploy it now to run through the tests, thanks!

**Comment by kitsirota** [comment]
@jhunt this looks good to me, could you review/merge this PR when you have a moment?

**Comment by jhunt** [comment]
Can you summarize the changes for a squashed merge commit message?

**Comment by wildloop** [comment]
I hope I did it correctly

**Comment by jhunt** [comment]
I just need a small blurb for the merge commit message, explaining what exactly this change introduces...

**Comment by wildloop** [comment]
This is the fix that allows at the same time access to the registry: 
- port 443: full, with password
- port 444: read-only, without password
